terraform/modules/vpc/main.tf

```diff
@@ -11,53 +11,33 @@ resource "aws_vpc" "vpc" {
   enable_dns_support   = "true"
   enable_dns_hostnames = "true"
   tags = {
-    Name = "${terraform.workspace}-${var.VPC_MODULE_NAME}"
+    Name = "${terraform.workspace}-${var.NAME}"
   }
 }
 
 # Subnets
-resource "aws_subnet" "public-subnet-1" {
+resource "aws_subnet" "public-subnets" {
+  count = length(var.PUBLIC_SUBNET)
   vpc_id                  = aws_vpc.vpc.id
-  cidr_block              = var.PUBLIC_SUBNET[0]
+  cidr_block              = var.PUBLIC_SUBNET[count.index]
   map_public_ip_on_launch = "true"
-  availability_zone       = data.aws_availability_zones.available.names[0]
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
```

Comment:
@Nouman72884 this statement will crash, if I specifies subnets counts greater than availability zones. e.g If I specifies 7 subnets in US-EAST, then it will crash as there are only 6 Availability Zones in US-EAST.